add-Tab-for-1-screen

Bleivox opened the pull request:
- Добавил вехний стак с лейблом и табом с кнопками "Шаблоны" и "Шаги" и нижнюю линию под кнопками с переходом
- Добавил использование Snapkit
- Добавил два Viewcontroller для заглушки

Bleivox committed:
- Добавил вехний стак с лейблом и табом с кнопками "Шаблоны" и "Шаги" и нижнюю линию под кнопками с переходом
- Добавил использование Snapkit
- Добавил два Viewcontroller для заглушки

NightDiRaven commented on Shablonator/ViewConrollers/TabsViewController.swift:
```diff
@@ -0,0 +1,101 @@
+import SnapKit
+
+final class TabsViewController: UIViewController {
+
```
```suggestion
private let tabs: [{ title: string, view: UIViewController }] = [
{title: 'Templates', view: TemplatesViewController },
{title: 'Steps', view: StepsViewController}
]
  private lazy var tabs = UnderlineTabsView(items: self.tabs.map(i => i.title)

```

NightDiRaven commented on Shablonator/ViewConrollers/TabsViewController.swift:
```diff
@@ -0,0 +1,101 @@
+    // MARK: Switching
+
+    private func switchTo(_ index: Int) {
+        let next = (index == 0) ? templatesVC : stepsVC
```
let next = self.tabs[index].view

NightDiRaven commented on Shablonator/ViewConrollers/TabsViewController.swift:
```diff
@@ -0,0 +1,101 @@
+    // MARK: UI
+
+    private func setupHeader() {
+        // фон шапки + скругление верхних углов
```
Вынести отсюда

NightDiRaven commented on Shablonator.xcodeproj/project.pbxproj:
```diff
@@ -6,8 +6,14 @@
 	objectVersion = 77;
```
Проверить, должно ли быть в репозитории, возможно добавить гит игнор

NightDiRaven commented on Shablonator/ViewConrollers/StepsViewController.swift:
```diff
@@ -0,0 +1,19 @@
+import UIKit
+import Foundation
+import SnapKit
+
```
```
abstract class StepViewController {
 constructor (Tab struct tab) {
   self.title = tab.title
 }

 override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .clear
        let l = UILabel(); l.text = self.title; l.textColor = .label
        view.addSubview(l)
        self. renderView()
        l.snp.makeConstraints { $0.center.equalToSuperview() }
    }

   abstract renderView();
}


final class StepsViewController: UIViewController extends StepViewController  {
    renderView () {
       делаем что-то
    }
}```

Bleivox committed:
Изменил логику согласно рекомендациям.
С гитом всё в порядке, ничего не надо убирать
В Файле Stepviewcontroller пока оставил всё так же. Логика с abstract не работает в swift
